File: Concurs Inot Java/RepoAngajat.java
package me.repository;


import me.model.Angajat;
import me.model.Participant;
import me.model.Proba;
import me.model.Stiluri;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class RepoAngajat implements IRepoAngajat{
    private static final Logger logger = LogManager.getLogger();
    JdbcUtils jdbcUtils;

    public RepoAngajat(Properties props){
        logger.info("Initializing RepoAngajat with properties: {} ",props);
        jdbcUtils=new JdbcUtils(props);
    }


    @Override
    public List<Angajat> findAngajatiNume(String nume) { return null; }

    @Override
    public Angajat getUsernameParola(String username, String parola) {
        logger.info("Se cauta angajatul cu username-ul "+ username+" parola *******");
        Connection con=jdbcUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("select * from Angajati where username=? and parola=?")){
            preStmt.setString(1,username);
            preStmt.setString(2,parola);
            try(ResultSet result=preStmt.executeQuery()) {
                if (result.next()) {
                    int id = result.getInt("id");
                    String nume = result.getString("nume");
                    int varsta = result.getInt("varsta");
                    return new Angajat(id,nume,varsta,username,parola);
                }
            }
        }catch (SQLException ex){
            System.out.println("Error DB "+ex);
        }
        return null;
    }

    @Override
    public Angajat save(Angajat elem) {
        logger.info("Se conecteaza la baza de date");
        Connection con=jdbcUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("insert into Angajati (nume,varsta,username,parola) values (?,?,?,?)")){
            preStmt.setString(1,elem.getNume());
            preStmt.setInt(2,elem.getVarsta());
            preStmt.setString(3,elem.getUsername());
            preStmt.setString(4,elem.getParola());
            preStmt.executeUpdate();
        }catch (SQLException ex){
            logger.info("eroare baza de date: "+ex.getMessage());
        }
        logger.info("S-a adaugat in baza de date angajatul "+elem);
        return elem;
    }

    @Override
    public Angajat update(Angajat elem) { return null; }

    @Override
    public Angajat delete(Angajat elem) throws IllegalArgumentException {
        return null;
    }

    @Override
    public Angajat get(Integer integer) { return null; }

    @Override
    public int size() { return 0; }

    @Override
    public Iterable<Angajat> findAll() {
        /*logger.info("Se cauta angajatii");
        Connection con=jdbcUtils.getConnection();
        List<Angajat> angajati = new ArrayList<>();
        try(PreparedStatement preStmt=con.prepareStatement("select * from Angajati")){
            try(ResultSet result=preStmt.executeQuery()) {
                if (result.next()) {
                    int id = result.getInt("id");
                    String nume = result.getString("nume");
                    int varsta = result.getInt("varsta");
                    String username = result.getString("username");
                    String parola = result.getString("parola");
                    Angajat a = new Angajat(id,nume,varsta,username,parola);
                    angajati.add(a);
                }
                return angajati;
            }
        }catch (SQLException ex){
            System.out.println("Error DB "+ex);
        }
        return null;*/
        logger.info("Se cauta angajatii");
        Connection con=jdbcUtils.getConnection();
        List<Angajat> angajati = new ArrayList<>();
        try(PreparedStatement preStmt=con.prepareStatement("select * from Angajati")){
            try(ResultSet result=preStmt.executeQuery()) {
                while (result.next()) {
                    int id = result.getInt("id");
                    String nume = result.getString("nume");
                    int varsta = result.getInt("varsta");
                    String username = result.getString("username");
                    String parola = result.getString("parola");
                    angajati.add(new Angajat(id,nume,varsta,username,parola));
                }
                return angajati;
            }
        }catch (SQLException ex){
            System.out.println("Error DB "+ex);
        }
        return null;
    }
}

File: Concurs Inot C#/RepoAngajat.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Problema_3_CSharp.me.model;
using Problema_3_CSharp.me.repository;
using log4net;
using Mono.Data.Sqlite;

namespace Problema_3_CSharp.me.repository
{
    public class RepoAngajat : IRepoAngajat
    {
        private static readonly ILog log = LogManager.GetLogger("LoggerRepoAngajat");
        public Angajat Delete(Angajat elem)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Angajat> FindAll()
        {
            log.Info("Se incepe conectarea la bd si selectarea tuturor angajatilor");
            var con = DBUtils.getConnection();
            IList<Angajat> angajati = new List<Angajat>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Angajati";

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int id = dataR.GetInt32(0);
                        string nume = dataR.GetString(1);
                        int varsta = dataR.GetInt32(2);
                        string username = dataR.GetString(3);
                        string parola = dataR.GetString(4);
                        angajati.Add(new Angajat(id,nume,varsta,username,parola));
                    }
                }
            }
            return angajati;
        }

        public IEnumerable<Angajat> FindAngajatiDupaNume(string nume)
        {
            throw new NotImplementedException();
        }

        public Angajat Get(int id)
        {
            throw new NotImplementedException();
        }

        public Angajat GetAngajatUsernameParola(string username, string parola)
        {
            log.InfoFormat("Se cauta angajatul cu username-ul {0} si parola ********", username);
            var con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Angajati where username=@username and parola=@parola";
                var paramUsername = comm.CreateParameter();
                paramUsername.ParameterName = "@username";
                paramUsername.Value = username;
                comm.Parameters.Add(paramUsername);

                var paramParola = comm.CreateParameter();
                paramParola.ParameterName = "@parola";
                paramParola.Value = parola;
                comm.Parameters.Add(paramParola);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int id = dataR.GetInt32(0);
                        String nume = dataR.GetString(1);
                        int varsta = dataR.GetInt32(2);
                        Angajat a = new Angajat(id, nume, varsta, username, parola);
                        log.InfoFormat("S-a gasit {0}", a);
                        return a;
                    }
                }
            }
            log.InfoFormat("Nu s-a gasit niciun angajat");
            return null;
        }

        public Angajat Save(Angajat elem)
        {
            log.Info("Se incearca adaugarea");
            var con = DBUtils.getConnection();
            using (var comm = con.CreateCommand())
            {
                
                comm.CommandText = "insert into Angajati (nume,varsta,username,parola)  values (@nume, @varsta, @username, @parola)";
                var paramNume = comm.CreateParameter();
                paramNume.ParameterName = "@nume";
                paramNume.Value = elem.Nume;
                comm.Parameters.Add(paramNume);

                var paramVarsta = comm.CreateParameter();
                paramVarsta.ParameterName = "@varsta";
                paramVarsta.Value = elem.Varsta;
                comm.Parameters.Add(paramVarsta);

                var paramUsername = comm.CreateParameter();
                paramUsername.ParameterName = "@username";
                paramUsername.Value = elem.Username;
                comm.Parameters.Add(paramUsername);

                var paramParola = comm.CreateParameter();
                paramParola.ParameterName = "@parola";
                paramParola.Value = elem.Parola;
                comm.Parameters.Add(paramParola);

                var result = comm.ExecuteNonQuery();

                log.Info("S-a adaugat " + elem.ToString());
                return elem;
            }
        }

        public int Size()
        {
            throw new NotImplementedException();
        }

        public Angajat Update(Angajat elem)
        {
            throw new NotImplementedException();
        }
    }
}
